### PR TITLE
Add auto-publish workflow

### PR DESCRIPTION
**Pull request by azidar**
Basically taken from Treadle: https://github.com/chipsalliance/treadle/blob/master/.github/workflows/test.yml

### CODE REVIEW

**Comment by jackkoenig** [comment]
Mill compilation failed, we may need to add the SNAPSHOTs resolvers to the `build.sc`.

File: build.sc [hunk 1]
```diff
@@ -28,6 +29,11 @@ trait CommonModule extends CrossUnRootedSbtModule with PublishModule {
     )
   )
 
+  override def repositories = super.repositories ++ Seq(
+    MavenRepository("https://oss.sonatype.org/content/repositories/snapshots"),
+    MavenRepository("https://oss.sonatype.org/content/repositories/releases")
+  )
```
Comment (jackkoenig) [inline build.sc:35]: Why is this necessary?

Comment (chick) [reply build.sc:35]: mill sanity check was failing until I put this in, could not find firrtl-interperter snapshot was first error. It could of been something else that changed I suppose

**Review by chick — APPROVED** [review]
Looks good to me.
Finally passes tests

**Comment by ekiwi** [comment]
Seems like the publishing plugin is missing:
Screenshot from 2021-04-13 08-42-07
